From d644cb5c3f580e4c5acb5a1274dc851a7e332a83 Mon Sep 17 00:00:00 2001
From: Jose Lamego <jose.a.lamego@intel.com>
Date: Tue, 24 Apr 2018 14:47:36 -0500
Subject: [PATCH] do not set cursor

some functions such as creating new vm host are failing
due to the set_cursor function in Gdk crashing when unable
to find the default cursor name.

This change allows the affected virt-manager functions to
succesfully finalize by eliminating the cursor setting calls.
This change must be reverted once the root cause for the
failing cursor lookup is fixed.

Signed-off-by: Jose Lamego <jose.a.lamego@intel.com>
---
 virtManager/asyncjob.py  | 4 ++--
 virtManager/baseclass.py | 8 ++++----
 virtManager/uiutil.py    | 2 +-
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/virtManager/asyncjob.py b/virtManager/asyncjob.py
index a96212e0..772bd70e 100644
--- a/virtManager/asyncjob.py
+++ b/virtManager/asyncjob.py
@@ -294,8 +294,8 @@ class vmmAsyncJob(vmmGObjectUI):
 
         if not self.cancel_cb and self.show_progress:
             gdk_window = self.topwin.get_window()
-            gdk_window.set_cursor(
-                Gdk.Cursor.new_from_name(gdk_window.get_display(), "progress"))
+            #gdk_window.set_cursor(
+            #    Gdk.Cursor.new_from_name(gdk_window.get_display(), "progress"))
         self._bg_thread.start()
 
 
diff --git a/virtManager/baseclass.py b/virtManager/baseclass.py
index c2f1ecde..589ffff5 100644
--- a/virtManager/baseclass.py
+++ b/virtManager/baseclass.py
@@ -264,8 +264,8 @@ class vmmGObjectUI(vmmGObject):
     def set_finish_cursor(self):
         self.topwin.set_sensitive(False)
         gdk_window = self.topwin.get_window()
-        cursor = Gdk.Cursor.new_from_name(gdk_window.get_display(), "progress")
-        gdk_window.set_cursor(cursor)
+        #cursor = Gdk.Cursor.new_from_name(gdk_window.get_display(), "progress")
+        #gdk_window.set_cursor(cursor)
 
     def reset_finish_cursor(self, topwin=None):
         if not topwin:
@@ -275,5 +275,5 @@ class vmmGObjectUI(vmmGObject):
         gdk_window = topwin.get_window()
         if not gdk_window:
             return
-        cursor = Gdk.Cursor.new_from_name(gdk_window.get_display(), "default")
-        gdk_window.set_cursor(cursor)
+        #cursor = Gdk.Cursor.new_from_name(gdk_window.get_display(), "default")
+        #gdk_window.set_cursor(cursor)
diff --git a/virtManager/uiutil.py b/virtManager/uiutil.py
index d07c2a1e..9932d2ab 100644
--- a/virtManager/uiutil.py
+++ b/virtManager/uiutil.py
@@ -96,7 +96,7 @@ def set_list_selection_by_number(widget, rownum):
     selection = widget.get_selection()
 
     selection.unselect_all()
-    widget.set_cursor(path)
+    #widget.set_cursor(path)
     selection.select_path(path)
 
 
-- 
2.17.0

